src/Chapter2.hs

```diff
@@ -727,7 +737,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate x = concatMap  (\n -> replicate n n) x
```

Comment:
Now, after you've mastered eta-reduction, you can apply such a technique to other functions as well  🙂 
But your implementation is already great 💯